subprojects/execution/src/main/java/org/gradle/internal/execution/steps/ResolveChangesStep.java

```diff
@@ -40,7 +40,7 @@
 
 public class ResolveChangesStep<R extends Result> implements Step<CachingContext, R> {
     private static final String NO_HISTORY = "No history is available.";
-    private static final String VALIDATION_FAILED = "Validation failed.";
+    private static final String VALIDATION_FAILED = "Change tracking has been disabled due to validation failures to ensure correctness. Please consult deprecation warnings for more details.";
```

Comment:
Suggestion:

```
Change tracking has been disabled to ensure correctness. Please consult deprecation warnings for more details.
```

As discussed, the term/concept of “validation failures” isn't meaningful to users an potentially confusing. I don't think it adds anything here and can be safely removed.